src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileParser.cs

```diff
@@ -2,25 +2,46 @@
 #nullable enable
 
 using System;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Text;
-using System.Threading.Tasks;
-using System.Xml;
-using Uno.Extensions;
-using System.IO;
 using System.Reflection;
-using Uno.UI.SourceGenerators.XamlGenerator.XamlRedirection;
-using System.Text.RegularExpressions;
+using System.Xml;
 using Windows.Foundation.Metadata;
-using Uno.UI.SourceGenerators.XamlGenerator.Utils;
-using System.Diagnostics;
+using Uno.Extensions;
 using Uno.Roslyn;
+using Uno.UI.SourceGenerators.XamlGenerator.Utils;
+using Uno.UI.SourceGenerators.XamlGenerator.XamlRedirection;
+
 
 namespace Uno.UI.SourceGenerators.XamlGenerator
 {
 	internal class XamlFileParser
 	{
+		private struct CachedFile
```

Comment:
We also need to validate the actual hash of the file's contents, otherwise we will not get file updates. We can build the `GetHashCode`/`Equals` overrides based on `IncludeXamlNamespaces`, `ExcludeXamlNamespaces`, the file content hash and the file path so that we can use it to get the value from the dictionary.